File: bcwAllSpice.client/src/models/Recipe.js
export class Recipe {
  constructor(data) {
    this.id = data.id
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.title = data.title
    this.subtitle = data.subtitle || ""
    this.instructions = data.instructions
    this.img = data.img || "https://andersnoren.se/themes/koji/wp-content/themes/koji/assets/images/default-fallback-image.png"
    this.category = data.category
    this.creatorId = data.creatorId
    this.creator = data.creator
    this.favoriteId = data.favoriteId
    this.favoritees = data.favoritees || []
    this.favoriteeIds = data.favoriteeIds || []
  }
}

File: bcwAllSpice/Models/Recipe.cs
namespace bcwAllSpice.Models;

public class Recipe : IRepoItem<int> {
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  [Required]
  public string Title { get; set; }
  public string Subtitle { get; set; }
  public string Instructions { get; set; }
  public string Img { get; set; }
  [Required]
  public string Category { get; set; }
  [Required]
  public string CreatorId { get; set; }

  // populated info
  public Profile Creator { get; set; }
  public List<string> FavoriteeIds { get; set; }
  public List<Profile> Favoritees { get; set; }
}